[CI] PyArrow and Pandas conflict caused by numpy 1.20

Currently, on CI there is a conflict between pyarrow and pandas. ref https://issues.apache.org/jira/browse/ARROW-5379

```
pyarrow/table.pxi:1394: in pyarrow.lib.Table.from_pandas
    ???
/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/pyarrow/pandas_compat.py:588: in dataframe_to_arrays
    for c, f in zip(columns_to_convert, convert_fields)]
/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/pyarrow/pandas_compat.py:588: in <listcomp>
    for c, f in zip(columns_to_convert, convert_fields)]
/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/pyarrow/pandas_compat.py:574: in convert_column
    raise e
/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/site-packages/pyarrow/pandas_compat.py:568: in convert_column
    result = pa.array(col, type=type_, from_pandas=True, safe=safe)
pyarrow/array.pxi:292: in pyarrow.lib.array
    ???
pyarrow/array.pxi:79: in pyarrow.lib._ndarray_to_array
    ???
pyarrow/array.pxi:67: in pyarrow.lib._ndarray_to_type
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   pyarrow.lib.ArrowTypeError: ('Did not pass numpy.dtype object', 'Conversion failed for column salary with type float64')

pyarrow/error.pxi:107: ArrowTypeError
```

--- a/environment.yml
+++ b/environment.yml
@@ -7,6 +7,7 @@ dependencies:
 - ibis-framework  # TODO: require Ibis 2.0 when it's released
 - pymapd>=0.24
 - pyarrow>=0.15.0
+- numpy<1.20
 
 # dev
 - black

--- a/requirements.txt
+++ b/requirements.txt
@@ -2,6 +2,7 @@
 ibis-framework
 pymapd>=0.24
 pyarrow>=0.15.0
+numpy<1.20
 
 # dev
 black

